.github/workflows/clippy.yml

```diff
@@ -1,14 +1,18 @@
 name: Clippy
 
-on: push
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:
 
 jobs:
   clippy_check:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v3
       - run: rustup component add clippy
-      - uses: actions-rs/clippy-check@v1
-        with:
-          token: ${{ secrets.GITHUB_TOKEN }}
-          args: -p openmls --tests -- -D warnings
+      - run: cargo clippy -p openmls --tests -- -D warnings
```

Comment:
I think you need to specify something like ` --message-format=json` so that GH can parse it for annotations.

Reply:
Oh, good point. I'll make a follow-up.